subprojects/gradle/namespaced-resources-fixer/src/main/kotlin/com/avito/android/plugin/FileResourcesFixer.kt

```diff
@@ -0,0 +1,169 @@
+package com.avito.android.plugin
+
+import io.github.detekt.parser.KtCompiler
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtImportList
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+import java.io.File
+
+/**
+ * Replace unambiguous R class references to Application by libraries R classes
+ */
+internal class FileResourcesFixer {
```

Comment:
The main idea:
- Find unique resources in libraries
- Find references to a merged R class file
- Replace them by references to libraries' R classes
- Add missing imports